internal/host/validator.go

```diff
@@ -1270,6 +1298,50 @@ func ignitionContainsManagedNetworkingFiles(config *ignition_types.Config) bool
 	return false
 }
 
+// workerIgnitionKubeletKubeconfigServerIsIPAddress looks for the kubelet
+// kubeconfig file within an ignition config and checks whether the server of
+// the cluster it points to is an IP address or a domain name
+func workerIgnitionKubeletKubeconfigServerIsIPAddress(config *ignition_types.Config) bool {
+	const kubeletKubeconfigIgnitionPath string = "/etc/kubernetes/kubeconfig"
+
+	if !ignitionHasFile(config, kubeletKubeconfigIgnitionPath) {
+		return false
+	}
+
+	kubeconfigBytes, err := ignitionReadFile(config, kubeletKubeconfigIgnitionPath)
+	if err != nil {
+		return false
+	}
+
+	type KubeconfigClusterInternal struct {
+		Server string `json:"server"`
+	}
+	type KubeconfigCluster struct {
+		Cluster KubeconfigClusterInternal `json:"cluster"`
+	}
+	type Kubeconfig struct {
+		Clusters []KubeconfigCluster `json:"clusters"`
+	}
+	var kubeconfig Kubeconfig
+
+	err = yaml.Unmarshal(kubeconfigBytes, &kubeconfig)
+	if err != nil {
+		return false
+	}
```

Comment:
Why not use [KubectlConfig](https://github.com/kubernetes/kops/blob/v1.24.0/pkg/kubeconfig/config.go#L19)

Reply:
Seems overkill to depend on kops for something so simple